File: src/main/groovy/com/bugsnag/android/gradle/SourceControl.java
package com.bugsnag.android.gradle;

public class SourceControl {
    String provider = null;
    String repository = null;
    String revision = null;

    public String getProvider() {
        return provider;
    }

    public void setProvider(String provider) {
        this.provider = provider;
    }

    public String getRepository() {
        return repository;
    }

    public void setRepository(String repository) {
        this.repository = repository;
    }

    public String getRevision() {
        return revision;
    }

    public void setRevision(String revision) {
        this.revision = revision;
    }
}

File: src/main/kotlin/com/bugsnag/android/gradle/SourceControl.kt
package com.bugsnag.android.gradle

import org.gradle.api.model.ObjectFactory
import org.gradle.api.provider.Property
import javax.inject.Inject

open class SourceControl @Inject constructor(objects: ObjectFactory) {
    val provider: Property<String> = objects.property(String::class.java)
    val repository: Property<String> = objects.property(String::class.java)
    val revision: Property<String> = objects.property(String::class.java)
}
